src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopyISQLServerBulkRecordTest.java

```diff
@@ -150,6 +157,480 @@ public void testBulkCopyDateTimePrecision() throws SQLException {
+
+                // Perform the bulk copy operation
+                try (SQLServerBulkCopy bulkCopy = new SQLServerBulkCopy(connection)) {
+                    bulkCopy.setDestinationTableName(vectorTable);
+                    bulkCopy.addColumnMapping("varcharCol", "vectorCol");
+                    bulkCopy.writeToServer(resultSet);
+                }
+
+                // Validate the data in the destination table
+                String validateSql = "SELECT * FROM " + vectorTable;
+                try (ResultSet rs = statement.executeQuery(validateSql)) {
+                    assertTrue(rs.next(), "No data found in the destination table.");
+                    microsoft.sql.Vector resultVector = rs.getObject("vectorCol", microsoft.sql.Vector.class);
+                    assertNotNull(resultVector, "Retrieved vector is null.");
+                    assertEquals(3, resultVector.getDimensionCount(), "Dimension count mismatch.");
+                    assertEquals(microsoft.sql.Vector.VectorDimensionType.float32, resultVector.getVectorDimensionType(),
+                            "Vector dimension type mismatch.");
+                    assertArrayEquals(new Object[] { 1.0f, 2.0f, 3.0f }, resultVector.getData(), "Vector data mismatch.");
+                }
+
+            }
+
+        } catch (Exception e) {
+            fail("Test failed with unexpected exception: " + e.getMessage());
+        } finally {
+            // Cleanup: Drop the tables
+            try (Connection connection = DriverManager.getConnection(connectionString);
+                    Statement statement = connection.createStatement()) {
+                TestUtils.dropTableIfExists(varcharTable, statement);
+                TestUtils.dropTableIfExists(vectorTable, statement);
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Test bulk copy from a VECTOR source column to a VARCHAR destination column.
+     * The operation should succeed, and the data should be validated.
+     */
+    @Test
+    @Tag(Constants.vectorTest)
+    public void testBulkCopyVectorUsingBulkCopyDestinationAsVarchar() {
+        String vectorTable = TestUtils.escapeSingleQuotes(AbstractSQLGenerator.escapeIdentifier("testVectorTable"));
+        String varcharTable = TestUtils.escapeSingleQuotes(AbstractSQLGenerator.escapeIdentifier("testVarcharTable"));
+
+        try (Connection connection = DriverManager.getConnection(connectionString);
+                Statement statement = connection.createStatement()) {
+
+            // Create the source table with a VECTOR column
+            statement.executeUpdate("CREATE TABLE " + vectorTable + " (vectorCol VECTOR(3))");
+
+            // Insert sample data into the source table
+            Object[] data = new Object[] { 1.0f, 2.0f, 3.0f };
+            microsoft.sql.Vector vector = new microsoft.sql.Vector(data.length, microsoft.sql.Vector.VectorDimensionType.float32, data);

```

Comment:
Would this be more readable with an import? Actually, I see we import microsoft.sql.Vector now. Do we need the full name here? (I see the pattern repeated elsewhere in this file. Please address all, if needed.)